app/src/main/java/ch/epfl/sdp/game/Server.java

```diff
@@ -170,25 +172,39 @@ private void initEnemies() {
         //  -------------------------------------------
     }
 
-    private void initCoins() {
-        int amount = 10;
-        ArrayList<Coin> coins = Coin.generateCoinsAroundLocation(playerManager.getCurrentUser().getLocation(), amount);
+
+    /**
+     * Randomly initializes the coin around the player's location
+     */
+    private void initCoinsAndShelterAreas() {
```

Comment:
Is there a difference between generating this for the Client and for the Server? Maybe we could put it as only one method to avoid duplication

Reply:
I already did this for some methods for the Solo mode and the Multi Player mode, so maybe we can do it for this too